src/Compilers/Core/Portable/Diagnostic/LocalizableResourceString.cs

```diff
@@ -113,5 +113,23 @@ public override string ToString(IFormatProvider formatProvider)
                 (_formatArguments.Length > 0 ? string.Format(resourceString, _formatArguments) : resourceString) :
                 string.Empty;
         }
+
+        public override bool Equals(LocalizableString other)
```

Comment:
@AnthonyDGreen Can you please review public API changes to LocalizableString and LocalizableResourceString? Basically they now implement IEquatable<T> so that descriptors (and diagnostics) based on LocalizableString arguments can correctly support value equality.